fuse_constraints/include/fuse_constraints/absolute_orientation_3d_stamped_euler_constraint.h

```diff
@@ -0,0 +1,146 @@
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_CONSTRAINTS_ABSOLUTE_ORIENTATION_3D_STAMPED_EULER_CONSTRAINT_H
+#define FUSE_CONSTRAINTS_ABSOLUTE_ORIENTATION_3D_STAMPED_EULER_CONSTRAINT_H
+
+#include <fuse_core/constraint.h>
+#include <fuse_core/macros.h>
+#include <fuse_core/uuid.h>
+#include <fuse_variables/orientation_3d_stamped.h>
+#include <geometry_msgs/PoseWithCovariance.h>
+#include <geometry_msgs/Quaternion.h>
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include <array>
+#include <ostream>
+
+
+namespace fuse_constraints
+{
+
+/**
+ * @brief A constraint that represents either prior information about a 3D orientation, or a direct measurement of the
+ * 3D orientation.
+ *
+ * This constraint holds the measured 3D orientation and the measurement uncertainty/covariance.
+ */
+class AbsoluteOrientation3DStampedEulerConstraint : public fuse_core::Constraint
+{
+public:
+  SMART_PTR_DEFINITIONS(AbsoluteOrientation3DStampedEulerConstraint);
+
+  using Euler = fuse_variables::Orientation3DStamped::Euler;
+
+  /**
+   * @brief Create a constraint using a measurement/prior of a 3D orientation
+   *
+   * @param[in] orientation The variable representing the orientation components of the pose
+   * @param[in] mean        The measured/prior Euler orientations in the order specified in /p axes
+   * @param[in] covariance  The measurement/prior covariance
+   * @param[in] axes        Used to specify which of the Euler axes they want to include in the constraint,
+   *                        e.g. "{ Euler::ROLL, EULER::YAW }"
+   */
+  AbsoluteOrientation3DStampedEulerConstraint(
+    const fuse_variables::Orientation3DStamped& orientation,
+    const Eigen::VectorXd& mean,
+    const Eigen::MatrixXd& covariance,
+    const std::vector<Euler> &axes);
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~AbsoluteOrientation3DStampedEulerConstraint() = default;
+
+  /**
+   * @brief Read-only access to the measured/prior vector of mean values.
+   *
+   * Order is defined by the provided \p axes parameter. This mean() function deviates from all other
```

Comment:
I dislike this constraint working differently than the others. And don't worry, I remember that I'm the one that wrote it. I'm just curious which implementation you think is "wrong"? This basically returns the same thing that was provided as input. That feels "more right" than what is being done in the other constraints. Just possibly less useful.

If we keep it this way, we may want to add read-only access to the `axes` vector. That way the mean/cov can be correctly interpreted by someone who didn't create it.

Reply:
Agreed on all counts. I think it's more strange for the use to give it a mean and then get a different one back.